doc/en/user/source/production/java.rst

```diff
@@ -17,19 +17,31 @@ Tested:
+
+For best performance we recommend the use *Oracle JRE 8* (also known as JRE 1.8).
+
+.. Further speed improvements can be released using `Marlin renderer <https://github.com/bourgesl/marlin-renderer>`__ alternate renderer.
+
 As of GeoServer 2.0, a Java Runtime Environment (JRE) is sufficient to run GeoServer.  GeoServer no longer requires a Java Development Kit (JDK).
 
-Install native JAI and JAI Image I/O extensions
------------------------------------------------
+Install native JAI and ImageIO extensions
+-----------------------------------------
 
-The `Java Advanced Imaging API <http://java.sun.com/javase/technologies/desktop/media/>`_ (JAI) is an advanced image manipulation library built by Oracle.  GeoServer requires JAI to work with coverages and leverages it for WMS output generation. By default, GeoServer ships with the pure Java version of JAI, but **for best performance, install the native JAI version in your JDK/JRE**.
+The `Java Advanced Imaging API <http://www.oracle.com/technetwork/java/javase/tech/jai-142803.html>`_ (JAI) is an advanced image processing library built by Oracle.  GeoServer requires JAI to work with coverages and leverages it for WMS output generation. JAI performance is important for all raster processing, which is used heavily in both WMS and WCS to rescale, cut and reproject rasters.
 
-In particular, installing the native JAI is important for all raster processing, which is used heavily in both WMS and WCS to rescale, cut and reproject rasters. Installing the native JAI is also important for all raster reading and writing, which affects both WMS and WCS.  Finally, native JAI is very useful even if there is no raster data involved, as WMS output encoding requires writing PNG/GIF/JPEG images, which are themselves rasters.
+The `Java Image I/O Technology <http://docs.oracle.com/javase/6/docs/technotes/guides/imageio/index.html>`__ (ImageIO) is used for  raster reading and writing. ImageIO affects both WMS and WCS for reading raster data, and is very useful (even if there is no raster data involved) for WMS output as encoding is required when writing PNG/GIF/JPEG images.
 
-Native extensions are available for Windows, Linux and Solaris (32 and 64 bit systems).  They are, however, not available for OS X.
+By default, GeoServer ships with the "pure java" version of JAI, but **for better performance JAI and ImageIO are available as "java extensions" to be installed into your JDK/JRE**. Native JAI and ImageIO extensions are available for Windows, Linux and Solaris (32 and 64 bit systems).  They are, however, not available for OS X.
 
+.. warning:: These native installers include a copy of JAI and JAI ImageIO as a "java extension" in your JRE. This may conflict with the pure java copy of JAI and JAI ImageIO included in your GeoServer ``libs`` folder - producing "class cast exceptions" preventing your application server from starting GeoServer.
```

Comment:
I would clarify that they also install JARs in the JRE in addition to native libs.